packages/sdk-manager/src/SdkManager.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EVMAssetTransfer,
+  Environment,
+  EvmFee,
+  Fungible,
+  Transfer
+} from '@buildwithsygma/sygma-sdk-core';
+import { BaseProvider, TransactionRequest } from '@ethersproject/providers';
+import { Signer } from 'ethers';
+import { UnsignedTransaction } from '@ethersproject/transactions';
+import { SdkManagerState, SdkManagerStatus } from './types';
+
+export class SdkManager implements SdkManagerState {
+  assetTransfer: EVMAssetTransfer;
+  status: SdkManagerStatus;
+  transfer?: Transfer<Fungible>;
+  fee?: EvmFee;
+  approvalTxs?: UnsignedTransaction[];
+  depositTx?: UnsignedTransaction;
+
+  constructor() {
+    this.assetTransfer = new EVMAssetTransfer();
+    this.status = 'idle';
+  }
+
+  async checkSourceNetwork(provider: BaseProvider) {
+    const providerChainId = (await provider.getNetwork()).chainId;
+    const validEnvDomains = this.assetTransfer.config
+      .getDomains()
+      .map((domain) => domain.chainId);
+
+    if (!validEnvDomains.includes(providerChainId)) {
+      this.status = 'invalidSourceNetwork';
+    } else {
+      this.status = 'initialized';
+    }
+  }
```

Comment:
nit: negative logic adds a bit more cognitive load, so maybe it'd be a good idea to remove `!` in the condition and switch clauses 🤷